### PR TITLE
feat: add redis chart

### PR DESCRIPTION
**Pull request by ambaliyav**

### CODE REVIEW

File: charts/redis/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+apiVersion: v2
+name: redis
+description: A Helm chart to deploy the redis for dev and test environment
+type: application
+# HelmChart Version
+version: 1.0.1
```
Comment (krishnamadhavan) [inline charts/redis/Chart.yaml]: As this is the first version, let it be `1.0.0`

File: charts/redis/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application.
+appVersion: 5.0.5
+home: https://github.com/whoacademy/helm-charts/charts/redis
+maintainers:
+  - name: springdo
+  - name: gkmahendran09
+  - name: rh-jpoole
```
Comment (krishnamadhavan) [inline charts/redis/Chart.yaml]: You can remove and add only your GitHub name here.

File: charts/redis/templates/redis-dc.yaml [hunk 3]
```diff
@@ -0,0 +1,78 @@
+{{- if .Values.managed_redis.disabled }}
```
Comment (krishnamadhavan) [inline charts/redis/templates/redis-dc.yaml]: This if statement is not required here. If we want a managed DB, we won't even deploy this entire chart. We can remove it.

Comment (krishnamadhavan) [inline charts/redis/templates/redis-dc.yaml]: Can we rename the file to `DeploymentConfig.yaml` ?

File: charts/redis/templates/redis-pvc.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{{- if .Values.managed_redis.disabled }}
```
Comment (krishnamadhavan) [inline charts/redis/templates/redis-pvc.yaml]: This if statement is not required here. If we want a managed DB, we won't even deploy this entire chart. We can remove it.

File: charts/redis/Chart.yaml (hunk 1 above)
Comment (ambaliyav) [reply charts/redis/Chart.yaml]: Done, It's updated

File: charts/redis/templates/redis-pvc.yaml (hunk 4 above)
Comment (krishnamadhavan) [inline charts/redis/templates/redis-pvc.yaml]: Can we rename this file to `PersistentVolumeClaim.yaml` ?

File: charts/redis/Chart.yaml (hunk 2 above)
Comment (ambaliyav) [reply charts/redis/Chart.yaml]: It's updated

File: charts/redis/templates/redis-secret.yaml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+{{- if .Values.managed_redis.disabled }}
```
Comment (krishnamadhavan) [inline charts/redis/templates/redis-secret.yaml]: This if statement is not required here. If we want a managed DB, we won't even deploy this entire chart. We can remove it.

Comment (krishnamadhavan) [inline charts/redis/templates/redis-secret.yaml]: Can we rename this file to `Secret.yaml` ?

File: charts/redis/templates/redis-service.yaml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+{{- if .Values.managed_redis.disabled }}
```
Comment (krishnamadhavan) [inline charts/redis/templates/redis-service.yaml]: This if statement is not required here. If we want a managed DB, we won't even deploy this entire chart. We can remove it.

Comment (krishnamadhavan) [inline charts/redis/templates/redis-service.yaml]: Can we rename this file to `Service.yaml` ?

File: charts/redis/values.yaml [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+managed_redis:
+  # disabled: true - to use in-cluster database
+  # disabled: false - use managed RDBMS database
+  disabled: true
+  # values will be overridden by the config repo based on the namespace
+  # different values for database will be available in values-*.yaml file from the config repo
+  auth:
+    host: localhost
+    password: password
```
Comment (krishnamadhavan) [inline charts/redis/values.yaml]: This block is not required here. If we want a managed DB, we won't even deploy this entire chart. We can remove it.

File: charts/redis/templates/redis-dc.yaml (hunk 3 above)
Comment (ambaliyav) [reply charts/redis/templates/redis-dc.yaml]: Yup, It's updated

Comment (ambaliyav) [reply charts/redis/templates/redis-dc.yaml]: Done

File: charts/redis/templates/redis-pvc.yaml (hunk 4 above)
Comment (ambaliyav) [reply charts/redis/templates/redis-pvc.yaml]: Yup, It's updated

Comment (ambaliyav) [reply charts/redis/templates/redis-pvc.yaml]: Yup, It's updated

File: charts/redis/templates/redis-secret.yaml (hunk 5 above)
Comment (ambaliyav) [reply charts/redis/templates/redis-secret.yaml]: Yup, It's updated

File: charts/redis/templates/redis-dc.yaml (hunk 3 above)
Comment (ambaliyav) [reply charts/redis/templates/redis-dc.yaml]: Done

File: charts/redis/templates/redis-secret.yaml (hunk 5 above)
Comment (ambaliyav) [reply charts/redis/templates/redis-secret.yaml]: Yup, It's updated

File: charts/redis/templates/redis-service.yaml (hunk 6 above)
Comment (ambaliyav) [reply charts/redis/templates/redis-service.yaml]: Yup, It's updated

Comment (ambaliyav) [reply charts/redis/templates/redis-service.yaml]: Yup, It's updated

File: charts/redis/values.yaml (hunk 7 above)
Comment (ambaliyav) [reply charts/redis/values.yaml]: Yup, It's removed

**Review by krishnamadhavan — APPROVED** [review]

**Pull request merged**